ui/pages/confirmations/hooks/useSignatureEventFragment.ts

```diff
@@ -0,0 +1,34 @@
+import { useCallback } from 'react';
+import { generateSignatureUniqueId } from '../../../helpers/utils/metrics';
+import { updateEventFragment } from '../../../store/actions';
+import { useConfirmContext } from '../context/confirm';
+import { SignatureRequestType } from '../types/confirm';
+import { isSignatureTransactionType } from '../utils';
+
+import type { MetaMetricsEventFragment } from '../../../../shared/constants/metametrics';
+
+/**
+ * When a signature has been requested, there should be an event fragment created for it in
+ * {@see {@link app/scripts/lib/createRPCMethodTrackingMiddleware.js}.
+ * This hook method is used to update an existing signature event fragment for a signature confirmation.
+ */
+export const useSignatureEventFragment = () => {
+  const { currentConfirmation } = useConfirmContext() as {
+    currentConfirmation: SignatureRequestType;
+  };
+
+  const requestId = currentConfirmation?.msgParams?.requestId as number;
+  const fragmentId = requestId ? generateSignatureUniqueId(requestId) : null;
+
+  const updateSignatureEventFragment = useCallback(
+    async (fragmentPayload: Partial<MetaMetricsEventFragment>) => {
+      if (!isSignatureTransactionType(currentConfirmation) || !fragmentId) {
+        return;
+      }
+      updateEventFragment(fragmentId, fragmentPayload);
+    },
+    [fragmentId],
```

Comment:
Are we not missing a dependency for `currentConfirmation`?

And should we refine that dependency or are any subsequent re-renders safe? My understanding is signature data is much more static than transactions.

Reply:
But we at least need the dependency since the callback references `currentConfirmation` as it passes it to `isSignatureTransactionType`?

Reply:
thanks for the suggestion @matthewwalsh0! we moved the isSignatureTransactionType checkout outside of useCallback https://github.com/MetaMask/metamask-extension/pull/27043/commits/d2faff38b26f63b4cf182ce194dd4bf2bd8029de